doc/builders/packages/fish.section.md

```diff
@@ -0,0 +1,49 @@
+# Fish {#sec-fish}
+
+Fish is a "smart and user-friendly command line shell" with plugins.
+
+
+## Vendor Fish scripts {#sec-fish-vendor}
+
+Any package may ship its own Fish completions, configuration snippets, and
+functions. Those should be installed to
+`$out/share/fish/vendor_{completions,conf,functions}.d` respectively.
+
+When the `programs.fish.enable` and
+`programs.fish.vendor.{completions,config,functions}.enable` options from the
+NixOS Fish module are set to true, those paths are symlinked in the current
+system environment and automatically loaded by Fish.
+
+
+## Packaging Fish plugins {#sec-fish-plugins-pkg}
+
+While packages providing standalone executables belong to the top level,
+packages which have the sole purpose of extending Fish belong to the
+`fishPlugins` scope and should be registered in
+`pkgs/shells/fish/plugins/default.nix`.
+
+The `buildFishPlugin` utility function can be used to automatically copy Fish
+scripts from `$src/{completions,conf,conf.d,functions}` to the standard vendor
+installation paths. It also sets up the test environment so that the optional
+`checkPhase` is executed in a Fish shell with the Fish functions and other
+plugins specified in `checkFunctionPath` and `checkPlugins` respectively.
```

Comment:
Can you explain the difference between these two, e.g. when should I use one or the other (or both)?

Reply:
`checkPlugins` allows using other already packaged plugins while `checkFunctionPath` is used to load custom test functions that are local to the current package. I've clarified the paragrpah and amended.